### DISCUSSION TITLE
Offer self/cls as first parameter in completion for method

### DISCUSSION THREAD

**Discussion by brettcannon** [microsoft/pylance-release#2596] [Enhancement]
### Discussed in https://github.com/microsoft/vscode-python/discussions/16713

<div type='discussions-op-text'>

<sup>Originally posted by **scottschmidl** July 17, 2021</sup>
For Python, is there a way to have the self keyword offered in a class? If not there should be; writing self every time is for peasants</div>

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
